syntax = "proto3";
package hvs;

message Terminal{
	 int32 id = 1;
	 string ip = 2;
	 string name = 3;
	 bool tstatus = 4;
}

message TerminalArray{
	repeated Terminal terminal = 1;
}

message ChannelPolling{
	 int32 id = 1;
	 int32 saperateNumber = 2;
	 int32 style = 3;
	 int32 pollingTime = 4;
	 bool active = 5;
	 bool inuse = 6;
	 bool isPolling = 7;
	 repeated string terminalID = 8;
}

message TextStyle{
	int32 size = 1;
	int32 location = 2;
	int32 color = 3;
	bool showfps = 4;
}

message WrapperMessage{
	oneof msg{
		ChannelPolling  cpl= 3;
		TerminalArray ta = 4;
		Terminal tml = 5;
		int64 debugStatus = 6;
		int64 packetGrepperStatus = 7;
		TextStyle textstyle = 8;
	}
}
